### PR TITLE
Google doc error

### PR DESCRIPTION
**Pull request by solar464**
A stop gap measure to prevent people from printing from Google Docs until the reason for the formatting errors can be found.

Not sure if there's a better way to tell if a file is a Google Doc than inspecting the title.

### CODE REVIEW

File: modules/ocf_printhost/files/enforcer [hunk 1]
```diff
@@ -117,6 +117,16 @@ NOTIFY_JOB_ERROR = dedent("""\
...
 """)
 
+NOTIFY_PROHIBITED_FILE_TYPE = dedent("""\
+        I'm afraid I can't let you do that.
+        Please download your file as a PDF before downloading.
+        {file_type} files disagree with my software.\
+""")
+
+PROHIBITED_FILE_TYPES = [
+    ('Google Doc', lambda job: job.doc_name[-14:] == ' - Google Docs')
```
Comment (dkess) [inline modules/ocf_printhost/files/enforcer]: Use the `endswith` string method

File: modules/ocf_printhost/files/enforcer [hunk 2]
```diff
@@ -117,6 +117,16 @@ NOTIFY_JOB_ERROR = dedent("""\
         Please contact a staff member for assistance.\
 """)
 
+NOTIFY_PROHIBITED_FILE_TYPE = dedent("""\
+        I'm afraid I can't let you do that.
+        Please download your file as a PDF before downloading.
+        {file_type} files disagree with my software.\
```
Comment (dkess) [inline modules/ocf_printhost/files/enforcer]: I'd reword this to something like, "Unfortunately we don't support printing from Google Docs. Please download your file as PDF, then try again."

File: modules/ocf_printhost/files/enforcer [hunk 3]
```diff
@@ -164,6 +174,16 @@ def send_printer_mail(message, job, quo):
 
 def prehook(c, r, job):
     quo = quota.get_quota(c, job.user)
+
+    # A stop gap until we figure out why some file types aren't printing correctly
+    for file_type, prohibited in PROHIBITED_FILE_TYPES:
```
Comment (dkess) [inline modules/ocf_printhost/files/enforcer]: Do we expect to have more "prohibited file types"? If not, this kind of abstraction isn't really necessary. Just have an if statement for gdocs.

File: modules/ocf_printhost/files/enforcer (hunk 2 above)
Comment (solar464) [reply modules/ocf_printhost/files/enforcer]: Fair enough.

File: modules/ocf_printhost/files/enforcer (hunk 3 above)
Comment (solar464) [reply modules/ocf_printhost/files/enforcer]: Abizer mentioned there may be more file types, I figured I'd leave it extendable just in case.

Comment (abizer) [reply modules/ocf_printhost/files/enforcer]: it stands to reason that Google Docs is not the only thing that fails - does the same occur for printing straight from bCourses or Gmail or something else? this amount of abstraction is hardly enterprise-ready, I think it's safe to plan for a printing-unfriendly future with this. I'd call it `PROBLEMATIC_FILE_SOURCES` or something like that though.

File: modules/ocf_printhost/files/enforcer [hunk 4]
```diff
@@ -117,6 +117,15 @@ NOTIFY_JOB_ERROR = dedent("""\
         Please contact a staff member for assistance.\
 """)
 
+NOTIFY_PROHIBITED_FILE_TYPE = dedent("""\
+        Unfortunately we don't support printing {file_type}.
```
Comment (abizer) [inline modules/ocf_printhost/files/enforcer]: I'd just say "We've encountered problems printing documents from this source. Please try to download your document as a PDF and print it from the desktop instead of the web browser." or something like that

File: modules/ocf_printhost/files/enforcer [hunk 5]
```diff
@@ -164,6 +173,16 @@ def send_printer_mail(message, job, quo):
...
     quo = quota.get_quota(c, job.user)
+
+    # A stop gap until we figure out why some file types aren't printing correctly
+    for file_source, problematic in PROBLEMATIC_FILE_SOURCE:
+        if problematic(job):
+            msg = NOTIFY_PROBLEMATIC_FILE_SOURCE.format(
+                file_source=file_source,
+            )
+            r.publish(job.user, msg)
+            sys.exit(255)
```
Comment (kpengboy) [inline modules/ocf_printhost/files/enforcer]: Nit: some exit codes have special meanings. You may wish to consider picking one which would not clash with any of these.

Context: https://www.tldp.org/LDP/abs/html/exitcodes.html#EXITCODESREF

**Comment by kpengboy** [comment]
How long does it take for this notification to pop up after the user prints something?

**Comment by solar464** [comment]
When I tested it, the error notification usually pops up within a a second or two.

**Comment by cg505** [comment]
Is there something keeping this from getting merged?

**Comment by keur** [comment]
This does not work.
